### ISSUE TITLE
Handle "could not import module" errors gracefully

### ISSUE THREAD

**Issue by clebergnu** [avocado-framework/avocado#5459]
**Describe the bug**
Avocado does not handle "could not import module" errors very gracefully, with error messages that are quite cryptic.

**Steps to reproduce**
Write a valid `avocado-instrumented` test, but with an invalid import.  Example:

```python
from avocado import Test

import foo


class PassTest(Test):

    """
    Example test that passes.

    :avocado: tags=fast
    """

    def test(self):
        """
        A test simply doesn't have to fail in order to pass
        """
```

And run it:

```
$ avocado run examples/tests/passtest.py 
JOB ID     : 3fee9803715e414a16c3dcf1ddb9ff2f6dc6c0bd
JOB LOG    : /home/cleber/avocado/job-results/job-2022-08-11T10.24-3fee980/job.log
 (1/1) examples/tests/passtest.py:PassTest.test: STARTED
 (1/1) examples/tests/passtest.py:PassTest.test:  ERROR: Test.__init__() got an unexpected keyword argument 'run.results_dir' (0.01 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/cleber/avocado/job-results/job-2022-08-11T10.24-3fee980/results.html
JOB TIME   : 1.47 s
```

**Expected behavior**
Instead of "unexpected argument..." a more clear error message such as: "failed to import the file containing the test" or something similar. 

**Current behavior**

From original reporter @jnsnow:

```
(08/27) tests/protocol.py:Connect.testBadUNIX:  ERROR:
        Test.__init__() got an unexpected keyword argument 'run.results_dir'
        (0.01 s)
```

**System information (please complete the following information):**
 - OS: ```LSB Version:	:core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
Distributor ID:	Fedora
Description:	Fedora release 36 (Thirty Six)
Release:	36
Codename:	ThirtySix```
 - Avocado version: 5a0c5b2348da450397287a0954e4c335c0d590a9
 - Avocado installation method: git

**Comment by pevogam** [comment]
I am still getting this with most recent avocado version:
```
  (028/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_retry_times: STARTED
 (028/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_retry_times:  ERROR: Test.__init__() got an unexpected keyword argument 'run.results_dir' (0.04 s)
 (029/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_retry_status: STARTED
 (030/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_retry_status_stop: STARTED
 (029/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_retry_status:  ERROR: Test.__init__() got an unexpected keyword argument 'run.results_dir' (0.04 s)
 (030/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_retry_status_stop:  ERROR: Test.__init__() got an unexpected keyword argument 'run.results_dir' (0.04 s)
 (031/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_retry_invalid: STARTED
 (031/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_retry_invalid:  ERROR: Test.__init__() got an unexpected keyword argument 'run.results_dir' (0.04 s)
 (032/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_exit_code: STARTED
 (032/180) selftests/isolation/test_cartesian_graph.py:CartesianGraphTest.test_run_exit_code:  ERROR: Test.__init__() got an unexpected keyword argument 'run.results_dir' (0.05 s)
```
in the GH actions [job](https://github.com/intra2net/avocado-i2n/actions/runs/4809200765/jobs/8560125956?pr=27) despite tests passing in a local environment. Is there any way to workaround this cryptic message? I assume it is now caused by something other than a failed import?

**Comment by pevogam** [comment]
> Hi @pevogam, in the GH actions [job](https://github.com/intra2net/avocado-i2n/actions/runs/4809200765/jobs/8560125956?pr=27#step:6:77) you are using avocado 98, is that correct? This has been fixed in avocado 100. So maybe updating avocado in CI environment might help.

Thanks @richtja will try it and come back to you, hopefully the verbosity is improved there.

**Comment by pevogam** [comment]
Hi @richtja the reason I use 98.0 is because it is a proper synced release available for both Avocado and Avocado VT:
```
ERROR: Could not find a version that satisfies the requirement avocado-framework-plugin-vt==100.1 (from versions: 57.4, 76.0, 77.0, 78.0, 79.0, 80.0, 81.0, 82.0, 83.0, 85.0, 86.0, 87.0, 88.0, 89.0, 90.0, 91.0, 92.0, 93.0, 94.0, 98.0)
ERROR: No matching distribution found for avocado-framework-plugin-vt==100.1

Notice:  A new release of pip is available: 22.0.4 -> 23.1.2
Notice:  To update, run: pip install --upgrade pip
```
Note here I tried to download Avocado VT 100.1 which has a release tag but as you can see it doesn't have a PyPI release.

This being said using other methods to simply clone the most recent repos did not show any errors any longer so my CI runs might have been a false alarm after all. Let's close this again and reopen later on if any of us is able to confirm the same issues again on the most recent version. Thanks @richtja for double checking this. I will expect newer versions to be also be more verbose in case of problems like the ones above so let's see.

**Comment by richtja** [comment]
I am closing this, If this problem will occurs again, feel free to reopen in.
